Description: Replace AES_ctr128_encrypt call with CRYPTO_ctr128_encrypt
Author: Nicholas Guriev <guriev-ns@ya.ru>
Forwarded: not yet
Last-Update: 2017-07-06

--- a/VoIPController.cpp
+++ b/VoIPController.cpp
@@ -102,6 +102,7 @@ bool VoIPController::didInitWin32TimeSca
 #ifndef TGVOIP_USE_CUSTOM_CRYPTO
 #include <openssl/sha.h>
 #include <openssl/aes.h>
+#include <openssl/modes.h>
 #include <openssl/rand.h>
 
 void tgvoip_openssl_aes_ige_encrypt(uint8_t* in, uint8_t* out, size_t length, uint8_t* key, uint8_t* iv){
@@ -131,7 +132,7 @@ void tgvoip_openssl_sha256(uint8_t* msg,
 void tgvoip_openssl_aes_ctr_encrypt(uint8_t* inout, size_t length, uint8_t* key, uint8_t* iv, uint8_t* ecount, uint32_t* num){
 	AES_KEY akey;
 	AES_set_encrypt_key(key, 32*8, &akey);
-	AES_ctr128_encrypt(inout, inout, length, &akey, iv, ecount, num);
+	CRYPTO_ctr128_encrypt(inout, inout, length, &akey, iv, ecount, num, (block128_f) AES_encrypt);
 }
 
 voip_crypto_functions_t VoIPController::crypto={
